oci/spec_opts_linux.go

```diff
@@ -62,6 +74,26 @@ func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
 	}
 }
 
+func getDeviceOwnershipOverride(UIDGIDStr string) (*uint32, *uint32) {
```

Comment:
FYI.. if we start going down this path for anything but UID:GID format for device override.. then you have these patterns to consider for what it is possible to be overriding:
```
// generateUserString generates valid user string based on OCI Image Spec
// v1.0.0.
//
// CRI defines that the following combinations are valid:
//
// (none) -> ""
// username -> username
// username, uid -> username
// username, uid, gid -> username:gid
// username, gid -> username:gid
// uid -> uid
// uid, gid -> uid:gid
// gid -> error
//
// TODO(random-liu): Add group name support in CRI.
func generateUserString(username string, uid, gid *runtime.Int64Value) (string, error) {
	var userstr, groupstr string
	if uid != nil {
		userstr = strconv.FormatInt(uid.GetValue(), 10)
	}
	if username != "" {
		userstr = username
	}
	if gid != nil {
		groupstr = strconv.FormatInt(gid.GetValue(), 10)
	}
	if userstr == "" {
		if groupstr != "" {
			return "", errors.Errorf("user group %q is specified without user", groupstr)
		}
		return "", nil
	}
	if groupstr != "" {
		userstr = userstr + ":" + groupstr
	}
	return userstr, nil
}
```

Reply:
still I think uid:gid is good for prototyping for this annotation.. key is to keep it in sync with the k8s and cod discussions and the crio impl.

Reply:
put another way if.. we did this and error out.. then we would start failing to run containers that used to run unless they added this new annotation...

Reply:
@kzys what about a log stating we are falling back to current device ownership vs override for the container because uid:gid ownership was not specified in the securityprofile?

Reply:
I generally prefer to return an error to the caller, rather than logging. It is much clear. Checking a single log line in a busy server is tedious. Not all callers can access logs.

Regarding backward-compatibility concern, how about making `""` no-op?

Reply:
by log I mean debug log for this prototype only :-) Yeah the current behavior is not an error. Yes, I think the result of "" is a no-op they way it's coded so far.  Sure return error is fine so long as the caller is going to catch it.. I just meant don't fail the create container for this.

Reply:
your earlier idea of just passing in something that is less complicated than the "uid:gid" string is making a lot of sense ...

Reply:
The syntax is internal to containerd. I chose `uid:gid` string to make it the same as the other parameters and it might be easier to handle if we want to allow `ctr run` to set device permissions at some point. I'm fine with two ints as well.

Reply:
> Regarding backward-compatibility concern, how about making "" no-op?

the current code skips calling this function if `""`.

Reply:
I could check for "\d*:\d*" regexp before calling the func too.